File: Programação III/TrabalhoJava/src/Candidato.java
import java.util.Calendar;
import java.util.Date;
import java.text.SimpleDateFormat;

public class Candidato implements Comparable<Candidato> {
    private String nome;
    private String numero;
    private String votosNominais;
    private String situacao;
    private String nomeUrna;
    private String sexo;
    private Date dataNasc;
    private String destinoVoto;
    private String numeroPartido;
    private String siglaPartido;

    public Candidato(String[] dadosCandidato ){
        setNumero(dadosCandidato[0]);
        setVotosNominais(dadosCandidato[1]);
        setSituacao(dadosCandidato[2]);
        setNome(dadosCandidato[3]);
        setNomeUrna(dadosCandidato[4]);
        setSexo(dadosCandidato[5]);
        setDataNasc(dadosCandidato[6]);
        setDestinoVoto(dadosCandidato[7]);
        setNumeroPartido(dadosCandidato[8]);
    }



    public String getNome() {
        return this.nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getNumero() {
        return this.numero;
    }

    public void setNumero(String numero) {
        this.numero = numero;
    }

    public String getVotosNominais() {
        return this.votosNominais;
    }

    public void setVotosNominais(String votosNominais) {
        this.votosNominais = votosNominais;
    }

    public String getSituacao() {
        return this.situacao;
    }

    public void setSituacao(String situacao) {
        this.situacao = situacao;
    }

    public String getNomeUrna() {
        return this.nomeUrna;
    }

    public void setNomeUrna(String nomeUrna) {
        this.nomeUrna = nomeUrna;
    }

    public String getSexo() {
        return this.sexo;
    }

    public void setSexo(String sexo) {
        this.sexo = sexo;
    }

    public Date getDataNasc() {
        return this.dataNasc;
    }

    public Integer getIdade(Date dataHoje) {
        Calendar dataNascimento = Calendar.getInstance();  
        dataNascimento.setTime(this.dataNasc); 
        Calendar hoje = Calendar.getInstance();  
        hoje.setTime(dataHoje); 

        int idade = hoje.get(Calendar.YEAR) - dataNascimento.get(Calendar.YEAR); 

        if (hoje.get(Calendar.MONTH) < dataNascimento.get(Calendar.MONTH)) {
        idade--;  
        } 
        else 
        { 
            if (hoje.get(Calendar.MONTH) == dataNascimento.get(Calendar.MONTH) && hoje.get(Calendar.DAY_OF_MONTH) < dataNascimento.get(Calendar.DAY_OF_MONTH)) {
                idade--; 
            }
        }

        return idade;
    }

    public void setDataNasc(String dataNasc) {
        SimpleDateFormat formato = new SimpleDateFormat("dd/MM/yyyy"); 
        try {
            Date data = formato.parse(dataNasc);
            this.dataNasc = data;
        }catch(Exception e){
            System.out.println(e);
        }
        
    }

    public String getDestinoVoto() {
        return this.destinoVoto;
    }

    public void setDestinoVoto(String destinoVoto) {
        this.destinoVoto = destinoVoto;
    }

    public String getNumeroPartido() {
        return this.numeroPartido;
    }

    public void setNumeroPartido(String numeroPartido) {
        this.numeroPartido = numeroPartido;
    }

    public String getSiglaPartido() {
        return this.siglaPartido;
    }

    public void setSiglaPartido(String siglaPartido) {
        this.siglaPartido = siglaPartido;
    }

    public int comparaCandidatoNome(Candidato candidato){
        if(this.nome == candidato.nome){
            return 1;
        }else{
            return 0;
        }
    }


    @Override 
    public int compareTo(Candidato outroCandidato) { 
        if (Integer.parseInt(this.votosNominais) > Integer.parseInt(outroCandidato.votosNominais)) { 
            return -1; 
        } if (Integer.parseInt(this.votosNominais) < Integer.parseInt(outroCandidato.votosNominais)) { 
            return 1; 
        } 
        return this.dataNasc.compareTo(outroCandidato.getDataNasc());
    }

    @Override
    public String toString() {
        return this.nome + " / " + this.nomeUrna + " (" + this.siglaPartido + ", "+ this.votosNominais + ( Integer.parseInt(this.votosNominais)>1 ? " votos)" : " voto)");
    }

    public void imprimeCandidato(){
        System.out.println("Numero: " + this.numero);
        System.out.println("Votos_nominais: " + this.votosNominais);
        System.out.println("Situacao: " + this.dataNasc);
        System.out.println("Nome: " + this.nome);
        System.out.println("Nome_urna: " + this.nomeUrna);
        System.out.println("Sexo: " + this.sexo);
        System.out.println("Data_nasc: " + this.dataNasc);
        System.out.println("Destino_voto: " + this.destinoVoto);
        System.out.println("Numero_partido: " + this.numeroPartido);
        System.out.println("--------------------------");    
    }

    
}

File: Programação III/TrabalhoCpp/src/Candidato.cpp
#include<cstdio>
#include <iostream>
#include "Candidato.h"


Candidato::Candidato(const vector<string>& dadosCandidato ) {
    setNumero(dadosCandidato[0]);
    setVotosNominais(dadosCandidato[1]);
    setSituacao(dadosCandidato[2]);
    setNome(dadosCandidato[3]);
    setNomeUrna(dadosCandidato[4]);
    setSexo(dadosCandidato[5]);
    setDataNasc(dadosCandidato[6]);
    setDestinoVoto(dadosCandidato[7]);
    setNumeroPartido(dadosCandidato[8]);
}

bool Candidato::candidato_eleito() const{
    if(!this->situacao.compare("Eleito")){
        return true;
    }
    return false;
}

void Candidato::calcula_idade_e_armazena(const string& data_cand , const string& data_eleicao){
    int dia_cand = stoi(data_cand.substr(0,2));
    int mes_cand = stoi(data_cand.substr(3,2));
    int ano_cand = stoi(data_cand.substr(6,4));

    int dia_eleicao = stoi(data_eleicao.substr(0,2));
    int mes_eleicao = stoi(data_eleicao.substr(3,2));
    int ano_eleicao = stoi(data_eleicao.substr(6,4));

    int idade = 0;
    if (mes_eleicao > mes_cand || (mes_eleicao == mes_cand  && dia_eleicao >= dia_cand)) {
        idade = ano_eleicao - ano_cand;
    }else {
     idade = ano_eleicao - ano_cand - 1;
    }
    
    setIdade(idade);
}


void Candidato::print() const {
    cout << this->nome << " / " <<  this->nomeUrna << " (" << this->siglaPartido <<", " << this->votosNominais << " votos)";
}

void Candidato::println() const {
    this->print();
    cout << endl;
}



Candidato::~Candidato(){
}
